### PR TITLE
disttask: rename scheduler to TaskExecutor

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #48795

Problem Summary:
scheduler in disttask framework actually execute subtasks of task.
So TaskExecutor might be a better name.
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — COMMENTED** [review]
8/52 will review after conflict fixed

File: pkg/disttask/framework/taskexecutor/manager.go [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package scheduler
+package taskexecutor
```
[the remark this answers is not in the document]
Comment (ywqzzy) [reply pkg/disttask/framework/taskexecutor/manager.go:15]: IMHO, don't want to mix up with pkg/exeuctor

File: pkg/disttask/framework/framework_test.go [hunk 2]
```diff
@@ -226,17 +225,13 @@ func TestFrameworkSetLabel(t *testing.T) {
 }
 
 func TestMultiTasks(t *testing.T) {
-	defer dispatcher.ClearDispatcherFactory()
-	defer scheduler.ClearSchedulers()
```
Comment (D3Hunter) [inline pkg/disttask/framework/framework_test.go:230]: why removed?

**Review by D3Hunter — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**